portal/v2/src/ClusterDetailListComponents/Statistics/StatisticsComponent.tsx

```diff
@@ -0,0 +1,165 @@
+    DefaultPalette.themeLight,
+    DefaultPalette.themeLighter,
+    DefaultPalette.themeLighterAlt,
+    DefaultPalette.themePrimary,
+    DefaultPalette.themeSecondary,
+    DefaultPalette.themeTertiary
+  ]
+
+  const _onLegendClickHandler = (selectedLegend: string | null | string[]): void => {
+    if (selectedLegend !== null) {
+      console.log(`Selected legend - ${selectedLegend}`);
+    }
+  };
+
+  function StatisticsHelperComponent(): JSX.Element {   
+    useEffect(() => {
+      if (props.metrics.length > 0) {
+        const newPoints: ILineChartPoints[] = []
+        props.metrics.forEach((metric, i) => {          
+          var dataPoints: ILineChartDataPoint[] = []
+          metric.MetricValue.forEach(metricValue => {
+            let timeStamp = new Date(metricValue.timestamp)
+            let metricsTime = convertToUTC(timeStamp)
+            var data: ILineChartDataPoint = {
+              x: metricsTime, y: metricValue.value
+            }
+            dataPoints.push(data)
+          })
+
+          var lineChartPoint: ILineChartPoints = {
+            legend: metric.Name,
+            onLegendClick: _onLegendClickHandler,
+            data: dataPoints,
+            color: colors[i]
+          }
+          newPoints.push(lineChartPoint)
+        })
+        setPoints(newPoints)    
+      }
+    },[props.metrics])
+
+    useEffect(() => {   
+      setData({
+        chartTitle: 'Line Chart',
+        lineChartData: points,
+      });
+      (points.length > 0) ? setSpinnerVisible(false) : setSpinnerVisible(true)
+    }, [points])
+    
+    useEffect(() => {   
+      setSpinnerVisible(true)
+    }, [props.duration, props.endDate])
+
+    const rootStyle = { width: `${width}px`, height: `${height}px` };
+    const tokens = {
+      sectionStack: {
+        childrenGap: 10,
+      },
+      spinnerStack: {
+        childrenGap: 20,
+      },
+    };
+    const rowProps: IStackProps = { horizontal: false, verticalAlign: 'center' };
+    
+    return (
+    <Stack>
+      <StackItem>
+        {
+          spinnerVisible
+          ?
+          <Stack  {...rowProps} tokens={tokens.spinnerStack}>
+            <StackItem> 
+              <Spinner size={SpinnerSize.large} />
+            </StackItem>
+            <div style={rootStyle}>
+              <LineChart
+                data={data}
+                strokeWidth={2}
+                tickFormat={timeFormat}
+                height={height}
+                legendProps={{ canSelectMultipleLegends: true, allowFocusOnLegends: true }}
+              />
+            </div>
+          </Stack>
+          :
+          <div style={rootStyle}>
+            <LineChart
+              data={data}
+              strokeWidth={2}
+              tickFormat={timeFormat}
+              height={height}
+              width={width}
+              legendProps={{ canSelectMultipleLegends: true, allowFocusOnLegends: true }}
+            />
+          </div>
+        }
```

Comment:
I would split these into individual rendering consts or functions. It's not clear at a glance that the two dom trees are displayed dependent on `spinnerVisible`

Reply:
I created a function and used that there. Do check. I will be pushing the changes shortly.